### ISSUE TITLE
Some error messages should be complete

### ISSUE THREAD

**Issue by CiGit** [esbenp/prettier-vscode#55]
In some case, prettier throws error messages which are not ParseError.
Currently all error message are truncated and only the first line is shown which may not be enough to really understand what's going wrong.

* ParseError is currently handled with an error message and "Show" action. This action will scroll to the error location.
* OtherError show only the first line in an errorMessage. An example is in https://github.com/esbenp/prettier-vscode/issues/54#issuecomment-288115696

I think we should handle errors which have no .loc differently. It is not possible to print more than one line with `window.showErrorMessage`. An alternative for those may be to open an [outputChanel](https://code.visualstudio.com/docs/extensionAPI/vscode-api#OutputChannel).

**Comment by esbenp** [comment]
I am not sure I understand what "no .loc" means?

**Comment by CiGit** [comment]
Oh sorry, I wrote that code part, was easier for me ;-) 
By no `.loc` I mean error objects which do not have that property. Parse error give you that property.

I think having everything on one line would be difficult to understand. We may try but having:
```
Validation Error: 
Option "trailingComma" must be of type:
    boolean
  but instead received:
    string

  Example:
  {
    "trailingComma": false
  }
```
seems ugly to me.

**Comment by esbenp** [comment]
Okay, so what do you think - how do we differentiate between the two? Do we check for the `loc` property or use `instanceof` or similar?

And should we use `showErrorMessage` for `ParseError` and then output other errors to an outputChannel?

**Comment by CiGit** [comment]
I'm not sure we have access to the Error constructor to be able to `instance of`,  `.loc` *should* be an error we can point in the code. 
At least I've not seen any other error yet with a `.loc` which wasn't a ParseError but *may* depend on the parser used. Honestly I haven't tested with flow (and typescript ?)

What about showing every errors in an outputChannel ? I thinks it's too much for a ParseError but it would be consistent.

**Comment by CiGit** [comment]
fixed in 0.21.0
